blaze-client/src/test/scala/org/http4s/blaze/client/ClientTimeoutSuite.scala

```diff
@@ -108,100 +95,110 @@ class ClientTimeoutSuite extends Http4sSuite {
-      tail = mkConnection(RequestKey.fromRequest(req), tickWheel, idleTimeout = 1.second)
-      q <- Queue.unbounded[IO, Option[ByteBuffer]]
-      h = new QueueTestHead(q)
-      (f, b) = resp.splitAt(resp.length - 1)
-      _ <- (q.enqueue1(Some(mkBuffer(f))) >> d.get >> q.enqueue1(Some(mkBuffer(b)))).start
-      c = mkClient(h, tail, tickWheel)()
-      s <- c.fetchAs[String](req)
-    } yield s).intercept[TimeoutException]
-  }
+  tickWheelFixture.test("Idle timeout on slow request body before receiving response") {
+    tickWheel =>
+      // Sending request body hangs so the idle timeout will kick-in after 1s and interrupt the request
+      val body = Stream.emit[IO, Byte](1.toByte) ++ Stream.never[IO]
+      val req = Request(method = Method.POST, uri = www_foo_com, body = body)
+      val h = new SlowTestHead(Seq(mkBuffer(resp)), 3.seconds, tickWheel)
+      val c = mkClient(h, tickWheel)(idleTimeout = 1.second)
 
-  tickWheelFixture.test("Not timeout on only marginally slow POST body".flaky) { tickWheel =>
-    def dataStream(n: Int): EntityBody[IO] = {
-      val interval = 100.millis
-      Stream
-        .awakeEvery[IO](interval)
-        .map(_ => "1".toByte)
-        .take(n.toLong)
-    }
-
-    val req = Request[IO](method = Method.POST, uri = www_foo_com, body = dataStream(4))
+      c.fetchAs[String](req).intercept[TimeoutException]
+  }
 
-    val tail = mkConnection(RequestKey.fromRequest(req), tickWheel, idleTimeout = 10.second)
-    val (f, b) = resp.splitAt(resp.length - 1)
-    val h = new SeqTestHead(Seq(f, b).map(mkBuffer))
-    val c = mkClient(h, tail, tickWheel)(requestTimeout = 30.seconds)
+  tickWheelFixture.test("Idle timeout on slow request body while receiving response body".fail) {
+    tickWheel =>
+      // Sending request body hangs so the idle timeout will kick-in after 1s and interrupt the request.
+      // But with current implementation the cancellation of the request hangs (waits for the request body).
+      (for {
+        _ <- IO.unit
+        body = Stream.emit[IO, Byte](1.toByte) ++ Stream.never[IO]
+        req = Request(method = Method.POST, uri = www_foo_com, body = body)
+        q <- Queue.unbounded[IO, Option[ByteBuffer]]
+        h = new QueueTestHead(q)
+        (f, b) = resp.splitAt(resp.length - 1)
+        _ <- (q.enqueue1(Some(mkBuffer(f))) >> IO.sleep(3.seconds) >> q.enqueue1(
+          Some(mkBuffer(b))
+        )).start
+        c = mkClient(h, tickWheel)(idleTimeout = 1.second)
+        s <- c.fetchAs[String](req)
+      } yield s).intercept[TimeoutException]
+  }
 
-    c.fetchAs[String](req).assertEquals("done")
+  tickWheelFixture.test("Not timeout on only marginally slow request body".flaky) { tickWheel =>
+    // Sending request body will take 1500ms. But there will be some activity every 500ms.
+    // If the idle timeout wasn't reset every time something is sent, it would kick-in after 1 second.
+    // The chunks need to be larger than the buffer in CachingChunkWriter
```

Comment:
Could we make this more explicit in the test suite by using `chunkBufferMaxSize + N` instead of magic `2000000`?

Reply:
Done.